### PR TITLE
Java: Add PrintAst

### LINKED ISSUE #3955: Implement PrintAST for Java
This issue is about adding support to the CodeQL VS Code extension for viewing ASTs of Java files. See the parent issue https://github.com/github/vscode-codeql/pull/492.

### PR DESCRIPTION
**Pull request by joefarebrother**
Resolves #3955

### CODE REVIEW

File: java/ql/src/printAst.ql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @name Print AST
+ * @description Outputs a representation of a file's Abstract Syntax Tree. This
+ *              query is used by the VS Code extension.
+ * @id csharp/print-ast
```
Comment (aibaars) [inline java/ql/src/printAst.ql:5]:
```suggestion
 * @id java/print-ast
```

File: java/ql/src/semmle/code/java/Javadoc.qll [hunk 2]
```diff
@@ -89,6 +91,8 @@ class JavadocTag extends JavadocElement, JavadocParent, @javadocTag {
 
   /** Gets the text associated with this Javadoc tag. */
   override string getText() { result = this.getChild(0).toString() }
+
+  override string getAPrimaryQlClass() { result = "Javadoc" }
```
Comment (aibaars) [inline java/ql/src/semmle/code/java/Javadoc.qll:95]:
```suggestion
  override string getAPrimaryQlClass() { result = "JavadocTag" }
```

File: java/ql/src/semmle/code/java/PrintAst.ql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @name Print AST
+ * @description Outputs a representation of the Abstract Syntax Tree.
+ * @id csharp/print-ast
```
Comment (aibaars) [inline java/ql/src/semmle/code/java/PrintAst.ql:4]:
```suggestion
 * @id java/print-ast
```

File: java/ql/src/semmle/code/java/PrintAst.ql [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @name Print AST
```
Comment (aibaars) [inline java/ql/src/semmle/code/java/PrintAst.ql:2]: Is this file needed for anything?

**Review by aibaars — COMMENTED** [review]
I had a quick look through the code. Looks good in general.

File: java/ql/src/semmle/code/java/PrintAst.ql (hunk 4 above)
Comment (joefarebrother) [reply java/ql/src/semmle/code/java/PrintAst.ql:2]: This is the file that the qlref of the test cases points to

Comment (aibaars) [reply java/ql/src/semmle/code/java/PrintAst.ql:2]: Ah, yes of course. Quite useful indeed.

**Comment by joefarebrother** [comment]
I believe the failing test is due to some semmle-code tests depending on the old PrintAst.qll. 
Would the best course of action be to rename my PrintAst.qll, so the name can be used as a deprecated alias of PrettyPrintAst.qll?

File: java/ql/src/semmle/code/java/Import.qll [hunk 5]
```diff
@@ -48,6 +50,8 @@ class ImportOnDemandFromType extends Import {
   NestedType getAnImport() { result.getEnclosingType() = this.getTypeHoldingImport() }
 
   override string toString() { result = "import " + this.getTypeHoldingImport().toString() + ".*" }
+
+  override string getAPrimaryQlClass() { result = "ImportOnDemandFromTypw" }
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/Import.qll:54]:
```suggestion
  override string getAPrimaryQlClass() { result = "ImportOnDemandFromType" }
```

**Comment by aschackmull** [comment]
> Would the best course of action be to rename my PrintAst.qll, so the name can be used as a deprecated alias of PrettyPrintAst.qll?

I think the cleanest is to get the rename of the old `PrintAst.qll` done in a separate PR, or two PRs really, as you'll need one PR in this repo and one in the semmle-code repo and then merge the two synchronously.

**Review by aschackmull — APPROVED** [review]
